emscripten.py

```diff
@@ -471,11 +479,13 @@ def finalize_wasm(infile, memfile, DEBUG):
     args.append('--dwarf')
   stdout = building.run_binaryen_command('wasm-emscripten-finalize',
                                          infile=infile,
-                                         outfile=infile if modify_wasm else None,
+                                         outfile=outfile if modify_wasm else None,
                                          args=args,
                                          stdout=subprocess.PIPE)
   if modify_wasm:
     building.save_intermediate(infile, 'post_finalize.wasm')
+  elif infile != outfile:
+    shutil.copy(infile, outfile)
```

Comment:
It would be nice if we could just rename here, but I guess we don't know if `infile` was a temp file or not...

Reply:
Yes, exactly.